pallets/xcm-asset-manager/src/lib.rs

```diff
@@ -0,0 +1,448 @@
+		/// MultiLocation Hash changes. Here we use AssetId instead. Changed the amount of units we
+		/// are charging per execution second for a given asset
+		UnitsPerSecondChanged { asset_id: T::AssetId, units_per_second: u128 },
+		/// Removed all information related to an assetId
+		ForeignAssetRemoved { asset_id: T::AssetId, asset_type: T::ForeignAssetType },
+		/// Litentry: Purestake use AssetType as index. This might lead to security issue if
+		/// MultiLocation Hash changes. Here we use AssetId instead.
+		/// New Event gives the info about involved asset_id, removed asset_type, and the new
+		/// default asset_id and asset_type mapping after removal
+		SupportedAssetTypeRemoved {
+			asset_id: T::AssetId,
+			removed_asset_type: T::ForeignAssetType,
+			default_asset_type: T::ForeignAssetType,
+		},
+	}
+
+	/// Mapping from an asset id to asset type.
+	/// This is mostly used when receiving transaction specifying an asset directly,
+	/// like transferring an asset from this chain to another.
+	#[pallet::storage]
+	#[pallet::getter(fn asset_id_type)]
+	pub type AssetIdType<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AssetId, T::ForeignAssetType>;
+
+	/// Reverse mapping of AssetIdType. Mapping from an asset type to an asset id.
+	/// This is mostly used when receiving a multilocation XCM message to retrieve
+	/// the corresponding asset in which tokens should me minted.
+	#[pallet::storage]
+	#[pallet::getter(fn asset_type_id)]
+	pub type AssetTypeId<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::ForeignAssetType, T::AssetId>;
+
+	/// Litentry: Purestake use AssetType as index. This might lead to security issue if
+	/// MultiLocation Hash changes. Here we use AssetId instead. Stores the units per second for
+	/// local execution for a AssetType. This is used to know how to charge for XCM execution in a
+	/// particular asset
+	/// Not all assets might contain units per second, hence the different storage
+	#[pallet::storage]
+	#[pallet::getter(fn asset_id_units_per_second)]
+	pub type AssetIdUnitsPerSecond<T: Config> = StorageMap<_, Blake2_128Concat, T::AssetId, u128>;
+
+	/// Stores the tracker of foreign assets id that have been
+	/// created so far
+	#[pallet::storage]
+	#[pallet::getter(fn foreign_asset_tracker)]
+	pub type ForeignAssetTracker<T: Config> = StorageValue<_, T::AssetId, ValueQuery>;
+
+	/// Litentry: Purestake use AssetType as index. This might lead to security issue if
+	/// MultiLocation Hash changes. Here we use AssetId instead.
+	// Supported fee asset payments
+	#[pallet::storage]
+	#[pallet::getter(fn supported_fee_payment_assets)]
+	pub type SupportedFeePaymentAssets<T: Config> = StorageValue<_, Vec<T::AssetId>, ValueQuery>;
+
+	/// The storages for AssetMetadatas.
+	///
+	/// AssetMetadatas: map AssetId => Option<AssetMetadata>
+	#[pallet::storage]
+	#[pallet::getter(fn asset_metadatas)]
+	pub type AssetMetadatas<T: Config> =
```

Comment:
Can we have a consistent naming for map? That is, something like AssetIdMetadata (similar to previous map variables)

Reply:
Changed. New name AssetIdMetadatas